pkg/scheduler/scheduler_test.go

```diff
@@ -1263,6 +1264,393 @@ func TestEntryOrdering(t *testing.T) {
+			}).
+			ResourceGroup(
+				*utiltesting.MakeFlavorQuotas("on-demand").
+					Resource(corev1.ResourceCPU, "50", "50").Obj(),
+				*utiltesting.MakeFlavorQuotas("spot").
+					Resource(corev1.ResourceCPU, "100", "0").Obj(),
+			).Obj(),
+		*utiltesting.MakeClusterQueue("eng-cohort-beta").
+			Cohort("cohort").
+			QueueingStrategy(kueue.StrictFIFO).
+			Preemption(kueue.ClusterQueuePreemption{
+				WithinClusterQueue:  kueue.PreemptionPolicyNever,
+				ReclaimWithinCohort: kueue.PreemptionPolicyLowerPriority,
+			}).
+			FlavorFungibility(kueue.FlavorFungibility{
+				WhenCanPreempt: kueue.Preempt,
+				WhenCanBorrow:  kueue.Borrow,
+			}).
+			ResourceGroup(
+				*utiltesting.MakeFlavorQuotas("on-demand").
+					Resource(corev1.ResourceCPU, "50", "50").Obj(),
+				*utiltesting.MakeFlavorQuotas("spot").
+					Resource(corev1.ResourceCPU, "100", "0").Obj(),
+			).Obj(),
+		*utiltesting.MakeClusterQueue("eng-cohort-theta").
+			Cohort("cohort").
+			QueueingStrategy(kueue.StrictFIFO).
+			Preemption(kueue.ClusterQueuePreemption{
+				WithinClusterQueue:  kueue.PreemptionPolicyNever,
+				ReclaimWithinCohort: kueue.PreemptionPolicyLowerPriority,
+			}).
+			FlavorFungibility(kueue.FlavorFungibility{
+				WhenCanPreempt: kueue.TryNextFlavor,
+				WhenCanBorrow:  kueue.TryNextFlavor,
+			}).
+			ResourceGroup(
+				*utiltesting.MakeFlavorQuotas("on-demand").
+					Resource(corev1.ResourceCPU, "50", "50").Obj(),
+				*utiltesting.MakeFlavorQuotas("spot").
+					Resource(corev1.ResourceCPU, "100", "0").Obj(),
+			).Obj(),
+	}
+
+	queues := []kueue.LocalQueue{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "main",
+			},
+			Spec: kueue.LocalQueueSpec{
+				ClusterQueue: "eng-alpha",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "main-alpha",
+			},
+			Spec: kueue.LocalQueueSpec{
+				ClusterQueue: "eng-cohort-alpha",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "main-beta",
+			},
+			Spec: kueue.LocalQueueSpec{
+				ClusterQueue: "eng-cohort-beta",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "main-theta",
+			},
+			Spec: kueue.LocalQueueSpec{
+				ClusterQueue: "eng-cohort-theta",
+			},
+		},
+	}
+	wl := utiltesting.MakeWorkload("low-1", "default").
+		Request(corev1.ResourceCPU, "50").
+		ReserveQuota(utiltesting.MakeAdmission("eng-alpha").Assignment(corev1.ResourceCPU, "on-demand", "50").Obj()).
+		Admitted(true).
+		Obj()
+	cases := []struct {
+		name                           string
+		cqs                            []kueue.ClusterQueue
+		admittedWorkloads              []kueue.Workload
+		workloads                      []kueue.Workload
+		deletedWorkloads               []kueue.Workload
+		wantPreempted                  sets.Set[string]
+		wantAdmissionsOnFirstSchedule  map[string]kueue.Admission
+		wantAdmissionsOnSecondSchedule map[string]kueue.Admission
```

Comment:
Based on what we are comparing below
```suggestion
		wantScheduledOnFirstSchedule  map[string]kueue.Admission
		wantAdmissionsAfterSecondSchedule map[string]kueue.Admission
```